audittools/trail.go

```diff
@@ -98,3 +87,24 @@ func (t AuditTrail) Commit(rabbitmqURI, rabbitmqQueueName string) {
 		}
 	}
 }
+
+func (rc *RabbitConnection) refresh(uri, queueName string) {
+	if rc != nil {
+		rc.Disconnect()
+	}
+
+	new, err := NewRabbitConnection(uri, queueName)
+	if err != nil {
+		logg.Error(err.Error())
+		return
+	}
+
+	*rc = *new
```

Comment:
This will break if `rc == nil` (which, if I understand the control flow correctly, is a very real possibility). I still think it would be much easier to just return a new `*RabbitConnection` and store it in `rc` on the caller side.